python/src/ops.cpp

```diff
@@ -3194,4 +3194,43 @@ void init_ops(py::module_& m) {
         Returns:
           result (array): The dequantized version of ``w``
       )pbdoc");
+  m.def(
+      "tensordot",
+      [](const array& a,
+         const array& b,
+         const std::variant<std::monostate, int, std::vector<std::vector<int>>>&
```

Comment:
It looks great, maybe a quick thing. Why do we need the monostate here? The argument is never `None` so it should be fine to have `std::variant<int, std::vector<std::vector<int>>>` ...

Reply:
Thanks for the clarification on the usage of monostate, will remove it from the variant list/if check as it isn’t needed